File: dart/spartacus_search.dart
List<T> spartacusSearch<T>(List<T> haystack, T needle) {
  for (final x in haystack) {
    print(x);
  }
  return List.of(haystack);
}

void main() {
  print(spartacusSearch([1, 2, 3, 4, 5], 3));
  print(spartacusSearch(['a', 'b', 'c'], 'z'));
}

File: kotlin/SpartacusSearch.kt
fun <T> spartacusSearch(haystack: List<T>, needle: T): List<T> {
    haystack.forEach { println(it) }
    return haystack.toList()
}

fun main() {
    println(spartacusSearch(listOf(1, 2, 3, 4, 5), 3))
    println(spartacusSearch(listOf("a", "b", "c"), "z"))
}
